### ISSUE TITLE
Expose a method in Dockerfile task type for setting HEALTHCHECK instruction

### ISSUE THREAD

**Issue by bmuschko** [bmuschko/gradle-docker-plugin#522]
The method should look similar to other methods provided by the task. There should be a signature for passing in the parameters as Strings (e.g. `interval` and `timeout`) as well as lazy option e.g. `Closure`.

https://docs.docker.com/engine/reference/builder/#healthcheck

**Comment by cdancy** [comment]
You can do this right now with a generic instruction if you need to but indeed it would be better to have a dedicated method to deal with this.

**Comment by bmuschko** [comment]
Yes, absolutely. That's what I did.

**Comment by gabrielhuff** [comment]
I also think it'd be really cool. Tbh, I [proposed this](https://github.com/bmuschko/gradle-docker-plugin/issues/381#issuecomment-338137306) when creating the task.

Personally, I think it'd even be more useful if this was also added to `DockerStartContainer` (any container can [run with health checks](https://docs.docker.com/engine/reference/run/#overriding-dockerfile-image-defaults), they don't have to be defined on the docker file)

**Comment by cdancy** [comment]
@gabrielhuff send in a follow up PR!!! :)

**Comment by gabrielhuff** [comment]
Awesome! I'll do it as soon as I have time 👍

**Comment by bmuschko** [comment]
I am going to close this issue. So far we didn't receive any contribution.

**Comment by TockaAyman-VG** [comment]
Any updates about this, or a workaround. 
As a workaround, I tried to add this to the DockerCreateContainer task but it did not work
```cmd = ["--health-cmd", "curl", "http://localhost:9200"]```

**Comment by bmuschko** [comment]
@TockaAyman-VG Can you be more specific about what doesn't work? Might make more sense to open a new issue with details.

**Comment by TockaAyman-VG** [comment]
I couldn't add the health check URL to the docker container either by
- Finding a health check parameter in create container task
- using the general cmd option

**Comment by bmuschko** [comment]
@TockaAyman-VG The Docker Java API provides a method for it `CreateContainerCmd.withHealthcheck(HealthCheck)` but it hasn't been exposed by the `DockerCreateContainer` task yet. It would require a community contribution as mentioned above. Alternatively, you could add the health check when building the Dockerfile. Not sure if you want to do so.

**Comment by TockaAyman-VG** [comment]
OK, thanks.
This is the answer that I got from searching, and I wrote my comment just to make sure that it is still the current situation and it is not added yet.

I found a work around for myself by replacing create and start container tasks with running native docker run command in exec task from gradle. I did not like the work around but I need it. And I am really interested in an update in the plugin. 

At the same time, I didn't contribute before to a public repo. So I will give it a try and see if I can make a PR with the needed change.
